src/main/ruby/core/kernel.rb

```diff
@@ -221,6 +226,30 @@ def display(port=$>)
     port.write self
   end
 
+  def eval(str, a_binding=nil, file=nil, line=nil)
+    file = "(eval)" unless file
+    line = 0 unless line
+    str = str.to_str unless str.class == String
+    file = file.to_str unless file.class == String
+    line = line.to_i unless line.class == Fixnum
+    unless a_binding
+      receiver = self
+      a_binding = Truffle.invoke_primitive(:caller_binding)
+    else
+      unless a_binding.class == Binding
+        raise(TypeError, "Wrong argument type %s #{a_binding.class} (expected binding)")
```

Comment:
Small thing, but I usually don't put `(` around raise as I consider it a sort of keyword,